packages/expo-cli/src/commands/upload/submission-service/archive-source/ArchiveFileSource.ts

```diff
@@ -0,0 +1,295 @@
+  mode: SubmissionMode,
+  source: ArchiveFileBuildIdSource
+): Promise<string> {
+  const { owner, slug } = getAppConfig(source.projectDir);
+  let build: any;
+  try {
+    build = await StandaloneBuild.getStandaloneBuildById({
+      platform: source.platform,
+      id: source.id,
+      owner,
+      slug,
+    });
+  } catch (err) {
+    console.error(err);
+    throw err;
+  }
+
+  if (!build) {
+    log.error(`Couldn't find build for id ${source.id}`);
+    return getArchiveFileLocationAsync(mode, {
+      sourceType: ArchiveFileSourceType.prompt,
+      platform: source.platform,
+      projectDir: source.projectDir,
+    });
+  } else {
+    return build.artifacts.url;
+  }
+}
+
+async function handlePromptSourceAsync(
+  mode: SubmissionMode,
+  source: ArchiveFilePromptSource
+): Promise<string> {
+  const { sourceType: sourceTypeRaw } = await prompt({
+    name: 'sourceType',
+    type: 'list',
+    message: 'What would you like to submit?',
+    choices: [
+      { name: 'I have a url to the app archive', value: ArchiveFileSourceType.url },
+      {
+        name: "I'd like to upload the app archive from my computer",
+        value: ArchiveFileSourceType.path,
+      },
+      {
+        name: 'The lastest build from Expo Servers',
+        value: ArchiveFileSourceType.latest,
+      },
+      {
+        name: 'A build identified by a build id',
+        value: ArchiveFileSourceType.buildId,
+      },
+    ],
+  });
+  const sourceType = sourceTypeRaw as ArchiveFileSourceType;
+  switch (sourceType) {
+    case ArchiveFileSourceType.url: {
+      const url = await askForArchiveUrlAsync();
+      return getArchiveFileLocationAsync(mode, {
+        sourceType: ArchiveFileSourceType.url,
+        url,
+        platform: source.platform,
+        projectDir: source.projectDir,
+      });
+    }
+    case ArchiveFileSourceType.path: {
+      const path = await askForArchivePathAsync();
+      return getArchiveFileLocationAsync(mode, {
+        sourceType: ArchiveFileSourceType.path,
+        path,
+        platform: source.platform,
+        projectDir: source.projectDir,
+      });
+    }
+    case ArchiveFileSourceType.latest: {
+      return getArchiveFileLocationAsync(mode, {
+        sourceType: ArchiveFileSourceType.latest,
+        platform: source.platform,
+        projectDir: source.projectDir,
+      });
+    }
+    case ArchiveFileSourceType.buildId: {
+      const id = await askForBuildIdAsync();
+      return getArchiveFileLocationAsync(mode, {
+        sourceType: ArchiveFileSourceType.buildId,
+        id,
+        platform: source.platform,
+        projectDir: source.projectDir,
+      });
+    }
+    case ArchiveFileSourceType.prompt:
+      throw new Error('This should never happen');
+  }
+}
+
+async function askForArchiveUrlAsync(): Promise<string> {
+  const { url } = await prompt({
+    name: 'url',
+    message: 'URL:',
+    type: 'input',
+    validate: (url: string): string | boolean =>
```

Comment:
maybe we can make the default value here something like `https://path.to/your/archive.aab`